src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ContentService.cs

```diff
@@ -95,49 +93,70 @@ public async Task DeletePreviousVersionsDownloadFiles(params Guid[] releaseIds)
             }
         }
 
-        public async Task UpdateContent(PublishContext context, params Guid[] releaseIds)
+        public async Task UpdateContent(params Guid[] releaseIds)
         {
             var releases = (await _releaseService
-                .List(releaseIds))
+                    .List(releaseIds))
                 .ToList();
 
+            foreach (var release in releases)
+            {
+                await _releaseCacheService.UpdateRelease(
+                    release.Id,
+                    publicationSlug: release.Publication.Slug,
+                    releaseSlug: release.Slug);
+            }
+
             var publications = releases
                 .Select(release => release.Publication)
                 .DistinctByProperty(publication => publication.Id)
                 .ToList();
 
             foreach (var publication in publications)
             {
-                await CacheLatestRelease(publication, context, releaseIds);
-                foreach (var release in releases)
-                {
-                    await CacheRelease(release, context);
-                }
+                // Cache the latest release for the publication as a separate cache entry
+                var latestRelease = await _releaseService.GetLatestRelease(publication.Id, releaseIds);
+                await _releaseCacheService.UpdateRelease(
+                    latestRelease.Id,
+                    publicationSlug: publication.Slug);
             }
         }
 
-        public async Task UpdateCachedTaxonomyBlobs()
+        public async Task UpdateContentStaged(DateTime expectedPublishDate, params Guid[] releaseIds)
         {
-            await _methodologyCacheService.UpdateSummariesTree();
-            await _publicationCacheService.UpdatePublicationTree();
-        }
+            var releases = (await _releaseService
+                    .List(releaseIds))
+                .ToList();
 
-        private async Task CacheLatestRelease(Publication publication, PublishContext context, params Guid[] includedReleaseIds)
-        {
-            var latestRelease = await _releaseService.GetLatestRelease(publication.Id, includedReleaseIds);
-            await _releaseCacheService.UpdateRelease(context.Staging,
-                context.Published,
-                latestRelease.Id,
-                publication.Slug);
+            foreach (var release in releases)
+            {
+                await _releaseCacheService.UpdateReleaseStaged(
```

Comment:
Oh yeah I bet you were loving having to put these variations in!

Reply:
Ha ha, not sure if I've made this better or worse. I think it's a little bit clearer now, at least for understanding when the cache key will use the staging directory, with it now being a separate key!